archivist/sboms.py

```diff
@@ -166,3 +173,45 @@ def list(
                 query=query,
             )
         )
+
+    def publish(self, identity: str) -> SBOM:
+        """Publish SBOMt
+
+        Makes an SBOM public.
+
+        Args:
+            identity (str): identity of SBOM
+
+        Returns:
+            :class:`SBOM` instance
+
+        """
+        LOGGER.debug("Publish SBOM %s", identity)
+        return SBOM(
+            **self._archivist.post(
```

Comment:
Type hint for `archivist.post` should be updated to say that `request` is an optional Dict